### PR TITLE
Rdm 12740

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
Author: it.consulting@tutanota.com <it.consulting@tutanota.com>
Date:   Fri Aug 27 16:07:53 2021 +0100

    RDM-12740: AC FTA - Create MVP Scenarios for Create Case
       [RDM-12707] (https://tools.hmcts.net/jira/browse/RDM-12740).

### CODE REVIEW

File: src/aat/resources/features/F-200 - New Access Control Create Case/S-200.1.td.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+  "_guid_": "S-200.1",
+
+  "productName": "CCD Data Store",
```
Comment (noronhaa) [inline src/aat/resources/features/F-200 - New Access Control Create Case/S-200.1.td.json:4]: Lets also create a F-200_CreateCase_Base file with the common lines of code and then have all the data files for each scenario extend that file

**Review by noronhaa — CHANGES_REQUESTED** [review]
Lets create a base file for the GetToken operation and also for the Create Case operation then have all the files that use those operation extend these new files to avoid some of the duplicated json lines

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/RoleAssignmentSetup/RoleAssignmentSetup.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    //         ArrayList<LinkedHashMap> roleAssingments = asRoleAssignemntCaseworker()
+    //            .get("/am/role-assignments/actors/82853c08-73dd-4bd3-a50e-7dae355da504")
+    //            .then()Befta_Default_Token_Creation_Data_For_Case_Creation
+    //            .extract().path("roleAssignmentResponse");
+    //
+    //         ArrayList<RoleAssignment> raList = (ArrayList<RoleAssignment>) roleAssingments.stream().map(ra ->
+    //             RoleAssignment.builder()
+    //             .roleName((String) ra.get("roleName"))
+    //             .roleType((String) ra.get("roleType"))
+    //             .grantType((String) ra.get("grantType"))
```
Comment (kiran-yenigala-hmcts) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/befta/RoleAssignmentSetup/RoleAssignmentSetup.java:37]: delete these commented lines if needed we can get from history

File: src/aat/resources/features/F-200 - New Access Control Create Case/F-200_CT1_GetToken.td.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
+{
```
[the remark this answers is not in the document]
Comment (Thor-tech-of-metal) [reply src/aat/resources/features/F-200 - New Access Control Create Case/F-200_CT1_GetToken.td.json:1]: DONE

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/RoleAssignmentSetup/RoleAssignmentSetup.java (hunk 2 above)
Comment (Thor-tech-of-metal) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/befta/RoleAssignmentSetup/RoleAssignmentSetup.java]: DONE

File: src/aat/resources/features/F-200 - New Access Control Create Case/S-200.4.td.json [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  "specs": [
+    "a role assignment to access CaseType CT2",
+    "attempts to create a case for CT2"
+  ],
+
+  "users" : {
+    "invokingUser": {
+      "username": "ccd.ac.solicitor1@gmail.com",
+      "password": "Pa55word11"
```
Comment (noronhaa) [inline src/aat/resources/features/F-200 - New Access Control Create Case/S-200.4.td.json:17]: lets hide the password with an env var, can just use [[$CCD_BEFTA_CASEWORKER_1_PWD]] for now until we create new env vars in vault and jenkins file

File: src/aat/resources/features/F-200 - New Access Control Create Case/S-200_Base.td.json [hunk 5]
```diff
@@ -0,0 +1,51 @@
+{
+  "_guid_": "S-200_Base",
+  "productName": "CCD Data Store",
+  "operationName": "create case",
+  "method": "POST",
+  "uri": "/case-types/{ctid}/cases",
+
+  "users" : {
+    "invokingUser": {
+      "username": "ccd.ac.solicitor1@gmail.com",
```
Comment (noronhaa) [inline src/aat/resources/features/F-200 - New Access Control Create Case/S-200_Base.td.json:10]: lets hide the password with an env var, can just use [[$CCD_BEFTA_CASEWORKER_1_PWD]] for now until we create new env vars in vault and jenkins file

**Review by noronhaa — APPROVED** [review]

File: src/aat/resources/features/F-200 - New Access Control Create Case/S-200_Base.td.json (hunk 5 above)
Comment (Thor-tech-of-metal) [reply src/aat/resources/features/F-200 - New Access Control Create Case/S-200_Base.td.json:10]: yes :) 👍  👍  I will sort it

**Pull request merged**